File: src/main/scala/com/ipponusa/strings/SimpleStringConsumer.scala
package com.ipponusa.strings

import java.util.Properties

import org.apache.kafka.clients.consumer.KafkaConsumer

import scala.collection.JavaConverters._

object SimpleStringConsumer extends App {

  val props = new Properties()
  props.put("bootstrap.servers", "localhost:9093")
  props.put("group.id", "mygroup")
  props.put("key.deserializer", "org.apache.kafka.common.serialization.StringDeserializer")
  props.put("value.deserializer", "org.apache.kafka.common.serialization.StringDeserializer")

  val topic = "mytopic"

  val consumer = new KafkaConsumer[String, String](props)
  consumer.subscribe(topic)

  while (true) {
    val records = consumer.poll(100).get(topic)
    for (record <- records.records().asScala) {
      println(s"key: ${record.key}")
      println(s"value: ${record.value}")
    }
  }

  consumer.close()

}

File: kafka-client-kotlin/src/main/kotlin/com/seigneurin/strings/SimpleStringConsumer.kt
package com.seigneurin.strings

import org.apache.kafka.clients.consumer.KafkaConsumer
import org.apache.kafka.common.serialization.StringDeserializer
import java.util.Properties

fun main(args: Array<String>) {
    val props = Properties()
    props["bootstrap.servers"] = "localhost:9092"
    props["group.id"] = "mygroup"
    props["key.deserializer"] = StringDeserializer::class.java.canonicalName
    props["value.deserializer"] = StringDeserializer::class.java.canonicalName

    val consumer = KafkaConsumer<String, String>(props)
    consumer.subscribe(listOf("mytopic"))

    val running = true
    while (running) {
        val records = consumer.poll(100)
        for (record in records) {
            println(record.value())
        }
    }

    consumer.close()
}
